src/Take.Blip.Builder/Variables/ResourceVariableProvider.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Lime.Protocol;
+using Lime.Protocol.Network;
+using Lime.Protocol.Serialization;
+using Take.Blip.Client;
+
+namespace Take.Blip.Builder.Variables
+{
+    public class ResourceVariableProvider : IVariableProvider
```

Comment:
This implementation is almost the same as BucketVariableProvider. 
Extract a base class for them (or reuse `BucketVariableProvider` in the same idea as made by the `ResourceExtension`)

Reply:
I created a base class for this implementation